tst/EnergyPlus/unit/PackagedTerminalHeatPump.unit.cc

```diff
@@ -4681,4 +4683,391 @@ TEST_F(EnergyPlusFixture, ZonePTHP_ElectricityRateTest)
+    state->dataHVACGlobal->TurnFansOn = false;
+    state->dataHVACGlobal->TurnFansOff = true;
+    state->dataHVACGlobal->ZoneCompTurnFansOff = false;
+    state->dataHVACGlobal->ZoneCompTurnFansOn = true;
+    state->dataHeatBalFanSys->TempZoneThermostatSetPoint(1) = 21.10;
+    state->dataHeatBalFanSys->TempTstatAir(1) = 21.1;
+    sysAvailMgr.AvailStatus = 0;
+    // run calc system availability requirement
+    SystemAvailabilityManager::CalcNCycSysAvailMgr(*state, SysAvailNum, PriAirSysNum, AvailStatus, ZoneEquipType, CompNum);
+    // check that the fan is off
+    EXPECT_EQ(DataHVACGlobals::NoAction, sysAvailMgr.AvailStatus);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOn);
+    EXPECT_TRUE(state->dataHVACGlobal->TurnFansOff);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOff);
+    EXPECT_TRUE(state->dataHVACGlobal->ZoneCompTurnFansOn);
+    // run to set zone night cycle manager
+    ZoneEquipmentManager::SimZoneEquipment(*state, true, SimAir);
+    Real64 OnOffAirFlowRatio = 0.0;
+    Real64 PartLoadRatio(1.0);
+    // now check if the global flag is reset correctly
+    thisSys.setAverageAirFlow(*state, PartLoadRatio, OnOffAirFlowRatio);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOff);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOff);
+
+    // test 2: availability manager status to on
+    state->dataHVACGlobal->ZoneComp(ZoneEquipType).ZoneCompAvailMgrs(1).StartTime = 0.0;
+    state->dataHVACGlobal->ZoneComp(ZoneEquipType).ZoneCompAvailMgrs(1).StopTime = 4.0;
+    state->dataHeatBalFanSys->TempZoneThermostatSetPoint(1) = 21.10;
+    state->dataHeatBalFanSys->TempTstatAir(1) = 21.5;
+    sysAvailMgr.AvailStatus = 0;
+    state->dataScheduleMgr->Schedule(1).CurrentValue = 1;
+    state->dataScheduleMgr->Schedule(2).CurrentValue = 0;
+    // run calc system availability requirement
+    SystemAvailabilityManager::CalcNCycSysAvailMgr(*state, SysAvailNum, PriAirSysNum, AvailStatus, ZoneEquipType, CompNum);
+    // check that the availability manager is cycling On
+    EXPECT_EQ(DataHVACGlobals::CycleOn, sysAvailMgr.AvailStatus);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOff);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOff);
+    state->dataHVACGlobal->ZoneComp(ZoneEquipType).ZoneCompAvailMgrs(1).AvailStatus = DataHVACGlobals::CycleOn;
+    // run to set zone night cycle manager
+    ZoneEquipmentManager::SimZoneEquipment(*state, true, SimAir);
+    // test global zone fan control variables are turned on
+    EXPECT_TRUE(state->dataHVACGlobal->TurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->TurnFansOff);
+    EXPECT_TRUE(state->dataHVACGlobal->ZoneCompTurnFansOn);
+    EXPECT_FALSE(state->dataHVACGlobal->ZoneCompTurnFansOff);
+}
+
```

Comment:
Added unit test that illustrates the `state->dataHVACGlobal->TurnFansOn` and `state->dataHVACGlobal->TurnFansOff` variables are reset correctly.